### DISCUSSION TITLE
RabbitTemplate - sendConnectionFactorySelectorExpression SpEL used for receive methods as well

### DISCUSSION THREAD

**Discussion by RedeemerSK** [spring-projects/spring-amqp#1414] [Q&A] [answered]
While looking at implementation of RabbitTemplate at https://github.com/spring-projects/spring-amqp/blob/d3b93e963e8ed784b6fb28a29b187af35d470405/spring-rabbit/src/main/java/org/springframework/amqp/rabbit/core/RabbitTemplate.java#L1075 I noticed that `sendConnectionFactorySelectorExpression` is always used inside `obtainTargetConnectionFactory()` regardless of what Expression is passed as parameter. Therefore, when called from `doReceive*()` methods, even though `receiveConnectionFactorySelectorExpression` is passed to `obtainTargetConnectionFactory()`, connection factory selection will be done based on evaluation of `sendConnectionFactorySelectorExpression` instead.

Seems like a bug. Or may it be intentional ?
(not creating bug ticket right away as I'm not yet an user of this module impacted by this implementation detail)

**Comment by garyrussell** [comment, marked as answer]
Thanks; yes, it's a bug. https://github.com/spring-projects/spring-amqp/issues/1415
